src/main/java/io/blockfrost/sdk/impl/ScriptServiceImpl.java

```diff
@@ -0,0 +1,142 @@
+        }
+    }
+
+    @Override
+    public ScriptJson getScriptJson(String scriptHash) throws APIException {
+        validateHash(scriptHash);
+
+        Call<ScriptJson> call = scriptsApi.scriptsScriptHashJsonGet(getProjectId(), scriptHash);
+
+        try {
+            Response<ScriptJson> response = call.execute();
+            return processResponse(response);
+        } catch (IOException exp) {
+            throw new APIException("Exception while fetching script JSON representation for hash: " + scriptHash, exp);
+        }
+    }
+
+    @Override
+    public ScriptCbor getScriptCbor(String scriptHash) throws APIException {
+        validateHash(scriptHash);
+
+        Call<ScriptCbor> call = scriptsApi.scriptsScriptHashCborGet(getProjectId(), scriptHash);
+
+        try {
+            Response<ScriptCbor> response = call.execute();
+            return processResponse(response);
+        } catch (IOException exp) {
+            throw new APIException("Exception while fetching script CBOR representation for hash: " + scriptHash, exp);
+        }
+    }
+
+    @Override
+    public List<ScriptRedeemer> getScriptRedeemers(String scriptHash, int count, int page, OrderEnum order) throws APIException {
+        ValidationHelper.validateCount(count);
+        validateHash(scriptHash);
+
+        Call<List<ScriptRedeemer>> call = scriptsApi.scriptsScriptHashRedeemersGet(getProjectId(), scriptHash, count, page, order.name());
+
+        try {
+            Response<List<ScriptRedeemer>> response = call.execute();
+            return processResponse(response);
+        } catch (IOException exp) {
+            throw new APIException("Exception while fetching script redeemers for hash: " + scriptHash, exp);
+        }
+    }
+
+    @Override
+    public ScriptDatum getScriptDatum(String datumHash) throws APIException {
+        validateHash(datumHash);
+
+        Call<ScriptDatum> call = scriptsApi.scriptsDatumDatumHashGet(getProjectId(), datumHash);
+
+        try {
+            Response<ScriptDatum> response = call.execute();
+            return processResponse(response);
+        } catch (IOException exp) {
+            throw new APIException("Exception while fetching script datum for hash: " + datumHash, exp);
+        }
+    }
+
+    @Override
+    public ScriptDatumCbor getScriptDatumCbor(String datumHash) throws APIException {
+        validateHash(datumHash);
+
+        Call<ScriptDatumCbor> call = scriptsApi.scriptsDatumDatumHashCborGet(getProjectId(), datumHash);
+
+        try {
+            Response<ScriptDatumCbor> response = call.execute();
+            return processResponse(response);
+        } catch (IOException exp) {
+            throw new APIException("Exception while fetching script datum CBOR representation for hash: " + datumHash, exp);
+        }
+    }
+
+    private static void validateHash(String hash) throws APIException {
```

Comment:
Should we move this to a util class to re-use in future if required?

Reply:
Moved to `ValidationHelper`